python/foglamp/services/core/api/filters.py

```diff
@@ -0,0 +1,245 @@
+    }'
+
+    Configuration item 'filter' is added to {service_name}
+    or updated with the pipeline arryay
+
+    Returns the filter pipeline on success:
+    {"pipeline": ["Scale10Filter", "Python_assetCodeFilter"]} 
+
+    Query string parameters:
+    - append_filter=true|false       Default true
+    - allow_duplicates=true|false    Default true
+
+    NOTE: the method also adds the filters category names under
+    parent category {service_name}
+"""
+
+async def add_filters_pipeline(request):
+    try:
+        # Get inout data
+        data = await request.json()
+        # Get filters list
+        filter_list = data.get('pipeline')
+        # Get filter name
+        service_name = request.match_info.get('service_name', None)
+        # Item name to add/update
+        config_item = "filter"
+
+        # Get configuration manager instance
+        cf_mgr = ConfigurationManager(connect.get_storage_async())
+        # Fetch the filter items: get category items
+        category_info = await cf_mgr.get_category_all_items(category_name = service_name)
+        if category_info is None:
+            # Error filter_name doesn't exist
+            message = "No such '%s' category found." % service_name
+            return web.HTTPNotFound(reason=message)
+
+        # Check whether config_item already exists
+        if (config_item in category_info):
+            # We just need to update the value of config_item
+            # with the "pipeline" property
+            # Check whether we want to replace or update the list
+            # or we allow duplicate entries in the list
+            # Default: append and allow duplicates
+            append_filter = 'true'
+            allow_duplicates = 'true'
+            if 'append_filter' in request.query and request.query['append_filter'] != '':
+                append_filter = request.query['append_filter'].lower()
+                if append_filter not in ['true', 'false']:
+                    raise ValueError("Only 'true' and 'false' are allowed for " \
+                                     "append_filter. {} given.".format(append_filter))
+            if 'allow_duplicates' in request.query and request.query['allow_duplicates'] != '':
+                allow_duplicates = request.query['allow_duplicates'].lower()
+                if allow_duplicates not in ['true', 'false']:
+                    raise ValueError("Only 'true' and 'false' are allowed for " \
+                                     "allow_duplicates. {} given.".format(allow_duplicates))
+
+            if (append_filter == 'true'):
+                # 'value' holds the string version of a list: convert it first  
+                current_value = json.loads(category_info[config_item]['value'])
+                # Save current list (ddepcopy)
+                new_list = copy.deepcopy(current_value['pipeline'])
+                # iterate inout filters list
+                for filter in filter_list:
```

Comment:
`Shadows built-in name 'filter'`, Use either for _filter

Reply:
fixed